socorro/processor/cache_manager.py

```diff
@@ -217,11 +220,21 @@ def make_room(self, size):
             total_size -= rm_size
             removed += rm_size
             try:
+                # Delete the evicted file
                 os.remove(rm_path)
             except FileNotFoundError:
                 # If there was an OSError, then this file is gone already. We need to
                 # update our bookkeping, so continue.
                 pass
+            try:
```

Comment:
This implements pruning empty directories which will in turn kick up DELETE | ISDIR events which will remove the corresponding watch. That'll prevent the cache manager from running out of inotify watches.